src/Core/maps/src/Maps/Handlers/Map/MapHandler.iOS.cs

```diff
@@ -0,0 +1,170 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Collections.Specialized;
+using System.ComponentModel;
+using CoreLocation;
+using MapKit;
+using Microsoft.Maui.Dispatching;
+using ObjCRuntime;
+using UIKit;
+using Microsoft.Maui.Handlers;
+using System.Drawing;
+using Microsoft.Maui.Maps.Platform;
+using Microsoft.Maui.Platform;
+using System.Runtime.CompilerServices;
+
+namespace Microsoft.Maui.Maps.Handlers
+{
+	
+	public partial class MapHandler : ViewHandler<IMap, MKMapView>
+	{
+		CLLocationManager? _locationManager;
+		UITapGestureRecognizer? _mapClickedGestureRecognizer;
+
+		protected override MKMapView CreatePlatformView()
+		{
+			return MapPool.Get() ?? new MauiMKMapView(this);
+		}
+
+		protected override void ConnectHandler(MKMapView platformView)
+		{
+			base.ConnectHandler(platformView);
+			_locationManager = new CLLocationManager();
+
+			PlatformView.AddGestureRecognizer(_mapClickedGestureRecognizer = new UITapGestureRecognizer(OnMapClicked));
+
+			if (platformView is MauiMKMapView customMKMapView)
+				customMKMapView.LayoutSubviewsFired += CustomMKMapViewLayoutSubviewsFired;
+
+			PlatformView.RegionChanged += MkMapViewOnRegionChanged;
+			//PlatformView.OverlayRenderer = GetViewForOverlay;
+
+			var mapsPinsItemsSource = (ObservableCollection<IMapPin>)VirtualView.Pins;
+			mapsPinsItemsSource.CollectionChanged += OnPinCollectionChanged;
+			OnPinCollectionChanged(mapsPinsItemsSource, new NotifyCollectionChangedEventArgs(NotifyCollectionChangedAction.Reset));
+		}
+
+		protected override void DisconnectHandler(MKMapView platformView)
```

Comment:
Assume none of this code will ever run. If there's code you really need to run when the mapview is no longer being used then you'll probably want to create a custom mapview and override "MovedtoWindow (sp?)"